hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java

```diff
@@ -292,7 +312,8 @@ private void addToObjectIdMap(Table<String, ? extends WithObjectID> fsTable,
         try {
           final WithObjectID oldKey = fsTable.get(key);
           final WithObjectID newKey = tsTable.get(key);
-          if (areKeysEqual(oldKey, newKey)) {
+          if (areKeysEqual(oldKey, newKey) || !isKeyInBucket(key, tablePrefixes,
```

Comment:
This is the core of the fix mentioned in the PR title, correct?

Next time, when possible, for a **bug** fix we can try to make the patch itself simple enough for review. Major refactoring and config additions can be done in separate PRs.

Reply:
For this PR, you should include the list of other changes in the first PR comment as well. Such as moving those Managed classes from framework to common, and the new force full diff config addition.

Reply:
Thanks @smengcl for the comment, I did not originally intend to make the change of moving these classes but it arose as a requirement while addressing one of the review comments, although will take a note to update the PR description every time such a change is made to simplify the review process and notify the reviewer.

Reply:
@sadanand48 Got it. That is alright. But let's include the extra changes in the PR description for record-keeping.